core/src/main/java/com/linecorp/armeria/client/HttpClientBuilder.java

```diff
@@ -35,6 +35,25 @@
  */
 public final class HttpClientBuilder extends AbstractClientOptionsBuilder<HttpClientBuilder> {
 
+    /**
+     * An undefined {@link URI} to create {@link HttpClient} without specifying {@link URI}.
+     */
+    private static final URI UNDEFINED_URI = URI.create("none+http://127.0.0.1");
+
+    /**
+     * Returns an undefined {@link URI}.
+     */
+    static URI undefinedUri() {
+        return UNDEFINED_URI;
```

Comment:
Probably OK to make `UNDEFINED_URI` package-local and accessing it directly, because it's an internal use only? I'm fine with as-is though.